src/cmap/connection_pool.ts

```diff
@@ -530,6 +537,12 @@ function createConnection(pool: ConnectionPool, callback?: Callback<Connection>)
   };
 
   pool[kPending]++;
+  // This is our version of a "virtual" no-I/O connection as the spec requires
```

Comment:
It would be more optimal to actually create a connection object here, so I'm open to other ideas, but all of the "better" approaches I can think of require an extensive refactor of the underlying connect method, that is several levels removed from the branching logic that calls different connection class constructors depending on the circumstances